hw/top_earlgrey/ip/ast/data/ast.hjson

```diff
@@ -36,42 +30,459 @@
   ],
   regwidth: "32",
   registers: [
-    { multireg:
-      {
-        name: "REGA",
-        desc: "AST Registers Array-A for OTP/ROM Write Testing",
-        count: "NumRegsA",
-        cname: "REGA",
-        swaccess: "rw",
-        hwaccess: "hro",
-        fields: [
-          { bits: "31:0",
-            name: "reg32",
-            desc: "32-bit Register",
-            resval: "0",
-          },
-        ],
-      },
+    { name: "REGA0",
```

Comment:
This expansion seems quite unnecessary,

Reply:
This is loop unrolling to enable more precise exclude function to the AST registers. Some of the AST registers AST are status only registers and some are write port. Since we can't reveal the registers from security reason, I need to have a more presice exclude description of the registers.

Reply:
You have your own internal `ast.hjson` which you use to create the correct chip RAL, which replaces the open source chip RAL. When you add those exclusions in your internal `ast.hjson`, they will show up in your internal chip RAL. There is no need for this change in the open source, since the open source chip RAL and your internal chip RAL models are completely unrelated.

Reply:
During the review of the ast registers model we were asked to expose as much as we can on those registers.
I started doing that by unrolling the loop and updating the exclude information. Next will be the access information.

Reply:
Like I said, this RAL and the RAL generated by you in NTIL are completely separate. Anything you do here has no effect on your internal extensions. I am really not sure how this is going to be any useful.